marlowe-chain-sync/libchainsync/Language/Marlowe/Runtime/ChainSync/QueryServer.hs

```diff
@@ -58,49 +55,46 @@ chainSyncQueryServer ChainSyncQueryServerDependencies{..} = ServerSource $ pure
             fmap (,serverReq) . traverseReqTree \case
               GetSecurityParameter -> queryGenesisParameters protocolParamSecurity
               GetNetworkId -> queryGenesisParameters protocolParamNetworkId
-              GetProtocolParameters -> queryShelley (const QueryProtocolParameters)
+              GetProtocolParameters -> queryShelley (\era -> QueryInShelleyBasedEraProjection QueryProtocolParameters $ fromLedgerPParams era)
```

Comment:
Eta reduce?

Reply:
Can't - it would result in impredicative type arguments to `(.)`

Reply:
Thanks for the explanation.